src/librustc_incremental/persist/dirty_clean.rs

```diff
@@ -73,17 +74,29 @@ pub fn check_dirty_clean_annotations<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
     let query = tcx.dep_graph.query();
     debug!("query-nodes: {:?}", query.nodes());
     let krate = tcx.hir.krate();
-    krate.visit_all_item_likes(&mut DirtyCleanVisitor {
+    let mut dirty_clean_visitor = DirtyCleanVisitor {
         tcx: tcx,
         query: &query,
         dirty_inputs: dirty_inputs,
-    });
+        checked_attrs: FxHashSet(),
+    };
+    krate.visit_all_item_likes(&mut dirty_clean_visitor);
+
+    let mut all_attrs = FindAllAttrs {
+        tcx: tcx,
+        attr_names: vec![ATTR_DIRTY, ATTR_CLEAN],
+        found_attrs: vec![],
+    };
+    intravisit::walk_crate(&mut all_attrs, krate);
+
+    all_attrs.report_unchecked_attrs(&dirty_clean_visitor.checked_attrs);
```

Comment:
can we add a comment explaining why we are not using the other "used attribute" system?